### PR TITLE
feat(schematics): Add ng-add support with prompt for making our schem…

### LINKED ISSUE #1538: Schematics: Add ng-add support with prompt for making our schematics default
`ng add @ngrx/schematics`

Requirements:

- [ ] Should prompt the user to set `@ngrx/schematics` as the default collection. If the user accepts, it should do the equivalent of https://ngrx.io/guide/schematics#default-schematics-collection
- [ ] Docs
- [ ] Tests

### PR DESCRIPTION
**Pull request by itayod**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

Closes #1538

## What is the new behavior?
Running `ng-add` will prompt the user to set `@ngrx/schematics` as the default collection.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by itayod** [comment]
@brandonroberts I hope that's what you had in mind :)

**Comment by brandonroberts** [comment]
@itayod thanks for working on this! What I had in mind is that this would be adding an `ng-add` to the `@ngrx/schematics` package similar to what the other packages have with a `schematics/ng-add` folder and a collection.

**Comment by itayod** [comment]
With pleasure! So just to make things clear, you want me to add `ng-add`  to `modules/schematics`?
Also, does the current functionality of setting `@ngrx/schematics` as default looks ok?

**Comment by brandonroberts** [comment]
Yes and yes

File: modules/schematics/collection.json [hunk 1]
```diff
@@ -2,52 +2,59 @@
...
       "schema": "./src/store/schema.json",
-      "description": "Adds initial setup for state managment"
+      "description": "Adds initial setup for state management"
+    },
+
+    "ng-add": {
+      "aliases": ["init"],
+      "factory": "./src/ng-add",
+      "schema": "./src/ng-add/schema.json",
+      "description": "Adds initial setup for Ngrx"
```
Comment (timdeschryver) [inline modules/schematics/collection.json]: What about using the following description here?

"Installs the NgRx schematics package"

File: modules/schematics-core/utility/config.ts [hunk 2]
```diff
@@ -145,3 +145,18 @@ export function getWorkspace(host: Tree): WorkspaceSchema {
 
   return JSON.parse(config);
 }
+
+export function setWorkspace(host: Tree, workSpace: WorkspaceSchema) {
+  const path = getWorkspacePath(host);
+  host.overwrite(path, JSON.stringify(workSpace, null, 2));
+}
+
+export function updateWorkspace(
```
Comment (timdeschryver) [inline modules/schematics-core/utility/config.ts]: This is only used within the @ngrx/schematics package.
Wouldn't it be better to define these functions within the package itself?

File: modules/schematics/src/ng-add/schema.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "id": "SchematicsNgRxRoot",
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.json]: Might be a bit weird, but the other packages are using the following convention:

```suggestion
  "id": "SchematicsNgRxSchematics",
```

File: modules/schematics/src/ng-add/schema.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "title": "NgRx Root Schema",
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.json]:
```suggestion
  "title": "Scaffolding library for Angular applications using NgRx libraries",
```

File: modules/schematics/src/ng-add/schema.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  "type": "object",
+  "properties": {
+    "setSchematicsDefault": {
+      "type": "boolean",
+      "default": true,
+      "description": "Make @ngrx/schematics as the default schematics",
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.json]:
```suggestion
      "description": "Use @ngrx/schematics as the default collection",
```

File: modules/schematics/src/ng-add/schema.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "id": "SchematicsNgRxRoot",
+  "title": "NgRx Root Schema",
+  "type": "object",
+  "properties": {
+    "setSchematicsDefault": {
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.json]: What about using just `default` or `defaultCollection` here?

File: modules/schematics/src/ng-add/schema.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export interface Schema {
+  setSchematicsDefault?: boolean;
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.ts]: Same here.

File: modules/schematics/src/ng-add/schema.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+      "type": "boolean",
+      "default": true,
+      "description": "Make @ngrx/schematics as the default schematics",
+      "x-prompt":
+        "Would you like to make @ngrx/schematics as the project's default?"
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.json]:
```suggestion
        "Would you want to use @ngrx/schematics as the default collection?"
```

File: modules/schematics/src/ng-add/index.spec.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    setSchematicsDefault: true,
+  };
+
+  let appTree: UnitTestTree;
+
+  beforeEach(() => {
+    appTree = createWorkspace(schematicRunner, appTree);
+  });
+
+  it('should set workspace default cli to @ngrx/schematics', () => {
```
Comment (timdeschryver) [inline modules/schematics/src/ng-add/index.spec.ts]: Should we also test on `setSchematicsDefault: false`?

File: modules/schematics/src/ng-add/schema.json (hunk 6 above)
Comment (timdeschryver) [inline modules/schematics/src/ng-add/schema.json]: What about adding an alias?

```
"aliases": ["d"]
```

**Comment by timdeschryver** [comment]
To not forget it, we should also install the @ngrx/schematics package.

**Comment by itayod** [comment]
> In order to not forget about it, we should also install the @ngrx/schematics package.

That's what I thought 😄

**Comment by brandonroberts** [comment]
We don't need to explicitly install the `@ngrx/schematics` package into the `package.json` anymore. You can do `ng add @ngrx/schematics` today and it will still add it to the project, but will fail when trying to run an ng-add schematic for it.

**Comment by brandonroberts** [comment]
You don't need to replace those. It will use NgRx schematics by default if the collection is set. We may do something with the defaults later, but that varies on the developer's project.

File: modules/schematics/src/ng-add/index.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import {
+  Rule,
+  SchematicContext,
+  Tree,
+  chain,
+  noop,
+} from '@angular-devkit/schematics';
+import { updateWorkspace } from '../../schematics-core/utility/config';
+import { Schema as RootStoreOptions } from './schema';
```
Comment (brandonroberts) [inline modules/schematics/src/ng-add/index.ts]:
```suggestion
import { Schema as SchematicOptions } from './schema';
```

File: modules/schematics/src/ng-add/index.spec.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import {
+  SchematicTestRunner,
+  UnitTestTree,
+} from '@angular-devkit/schematics/testing';
+import * as path from 'path';
+import { createWorkspace } from '../../../schematics-core/testing';
+import { Schema as RootStoreOptions } from './schema';
```
Comment (brandonroberts) [inline modules/schematics/src/ng-add/index.spec.ts]:
```suggestion
import { Schema as SchematicOptions } from './schema';
```

**Comment by brandonroberts** [comment]
Thanks again @itayod 👏

**Pull request merged**